### PR TITLE
test(typeorm): Add tests for typeorm

### PR DESCRIPTION
**Pull request by blessanabraham**
Copied over relevant tests from mongodb to typeorm.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -6,6 +6,12 @@ services:
       - '5432:5432'
     environment:
       POSTGRES_DB: accounts-js-tests-e2e
+  postgres_password:
```
Comment (pradel) [inline docker-compose.yml:9]: Would it be possible to reuse the e2e database image?

File: packages/database-typeorm/__tests__/typeorm-password.ts [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+      const services: any = retUser!.services;
+      expect(services.password[0].bcrypt).toBeTruthy();
+      expect(services.password[0].bcrypt).toEqual(newPassword);
+      expect(retUser!.createdAt).not.toEqual(retUser!.updatedAt);
+    });
+  });
+
+  describe('setUserDeactivated', () => {
+    // TODO: This should be added
+    // it('should throw if user is not found', async () => {
```
Comment (pradel) [inline packages/database-typeorm/__tests__/typeorm-password.ts:406]: Should this be done before merging?

File: packages/database-typeorm/__tests__/typeorm-sessions.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+describe('TypeormSessions', () => {
+  let userId: string;
+  let session: Record<string, string>;
+
+  let connection: Connection;
+  const url = 'postgres://postgres@localhost:5433/accounts-js-tests';
+
+  beforeAll(async () => {
+    // const adminConnection = await createConnection({
```
Comment (pradel) [inline packages/database-typeorm/__tests__/typeorm-sessions.ts:21]: can you please delete the commented code?

File: packages/database-typeorm/__tests__/typeorm-sessions.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+      expect(ret!.userAgent).toEqual('new user agent');
+      expect(ret!.valid).toEqual(true);
+      expect(ret!.token).toEqual(token);
+      expect(ret!.createdAt).toBeTruthy();
+      expect(ret!.updatedAt).toBeTruthy();
+      expect(ret!.createdAt).not.toEqual(ret!.updatedAt);
+    });
+
+    // TODO: updateSession does not accept a token like mongodb
+    // it('should update session with token', async () => {
```
Comment (pradel) [inline packages/database-typeorm/__tests__/typeorm-sessions.ts:162]: Should this be done before merging?

**Review by pradel — COMMENTED** [review]
Thank you very much for the pr, lack of tests was not good for this package :)

File: docker-compose.yml (hunk 1 above)
Comment (blessanabraham) [reply docker-compose.yml:9]: yes absolutely, I realised the tests were running in band after I created the PR. So there shouldn't be any conflicts with the e2e tests.

File: packages/database-typeorm/__tests__/typeorm-password.ts (hunk 2 above)
Comment (blessanabraham) [reply packages/database-typeorm/__tests__/typeorm-password.ts:406]: I actually added these tests because there is another bug I found in typeorm (will create an issue and PR for it) and the thought of making changes without any tests was scary. I wanted this PR to be only tests with no code changes whatsoever so that I can be sure things don't break. In the course of adding these tests from mongo, I also found a few other inconsistencies between the APIs. Will file bug reports for those as well.

TL;DR No, I want this to be just tests and will create separate PRs for other fixes. That commented out code is a reminder for me

File: packages/database-typeorm/__tests__/typeorm-sessions.ts (hunk 3 above)
Comment (blessanabraham) [reply packages/database-typeorm/__tests__/typeorm-sessions.ts:21]: sorry, my bad, will do

File: packages/database-typeorm/__tests__/typeorm-sessions.ts (hunk 4 above)
Comment (blessanabraham) [reply packages/database-typeorm/__tests__/typeorm-sessions.ts:162]: No, same as comment above, I wanted this PR to be only tests without any code changes. Will create separate PRs for other fixes

**Review by pradel — APPROVED** [review]
